src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -109,15 +116,8 @@ public class QueryProcessor implements QueryHandler
                              .executor(ImmediateExecutor.INSTANCE)
                              .maximumWeight(capacityToBytes(DatabaseDescriptor.getPreparedStatementsCacheSizeMiB()))
                              .weigher(QueryProcessor::getSizeOfPreparedStatementForCache)
-                             .removalListener((key, prepared, cause) -> {
-                                 MD5Digest md5Digest = (MD5Digest) key;
-                                 if (cause.wasEvicted())
-                                 {
-                                     metrics.preparedStatementsEvicted.inc();
-                                     lastMinuteEvictionsCount.incrementAndGet();
-                                     SystemKeyspace.removePreparedStatement(md5Digest);
-                                 }
-                             }).build();
+                             .removalListener((key, prepared, cause) -> evictPreparedStatement(key, cause))
```

Comment:
This may look a bit unnecessary, but needed this to instrument byteman to detect when this happens as `removePreparedStatement` is called in other code paths.  I think the code is marginally more concise/unharmfull in any case.